packages/build/src/steps/run_core_steps.ts

```diff
@@ -42,7 +41,7 @@ export const runCoreSteps = async (buildSteps: string[], flags: Partial<BuildFla
   }
 }
 
-const getBuildSteps = function (buildSteps: CoreStep[]) {
+const getBuildSteps = function (buildSteps: string[]) {
```

Comment:
This was incorrectly typed. `buildSteps` is an array of strings — example: https://github.com/netlify/cli/blob/a0f7247640a3ece1fc55702c4679796ccfcd642f/src/commands/deploy/deploy.ts#L373